src/broadcast/mpris/mprisplayer.cpp

```diff
@@ -0,0 +1,463 @@
+    return getVariantMapMetadata();
+}
+
+double MprisPlayer::volume() const {
+    return getAverageVolume();
+}
+
+void MprisPlayer::setVolume(double value) {
+    for (DeckAttributes* attrib : qAsConst(m_deckAttributes)) {
+        ControlProxy volume(ConfigKey(attrib->group, "volume"));
+        volume.set(value);
+    }
+}
+
+qlonglong MprisPlayer::position() const {
+    if (autoDjIdle()) {
+        for (unsigned int i = 0; i < m_pPlayerManager->numberOfDecks(); ++i) {
+            ControlProxy playing(ConfigKey(PlayerManager::groupForDeck(i), "play"));
+            if (playing.toBool()) {
+                DeckAttributes* pDeck = m_deckAttributes.at(i);
+                qlonglong playPosition =
+                        static_cast<qlonglong>(pDeck->playPosition() *   //Fraction of duration
+                                pDeck->getLoadedTrack()->getDuration() * //Duration in seconds
+                                1e6);
+                return playPosition;
+            }
+        }
+    }
+    return 0;
+}
+
+bool MprisPlayer::canGoNext() const {
+    return autoDjIdle();
+}
+
+bool MprisPlayer::canGoPrevious() const {
+    return false;
+}
+
+bool MprisPlayer::canPlay() const {
+    return true;
+}
+
+bool MprisPlayer::canPause() const {
+    return true;
+}
+
+bool MprisPlayer::canSeek() const {
+    return autoDjIdle();
+}
+
+void MprisPlayer::nextTrack() {
+    if (autoDjIdle()) {
+        m_pCPFadeNow->set(true);
+    }
+}
+
+void MprisPlayer::pause() {
+    if (autoDjIdle()) {
+        DeckAttributes* playingDeck = findPlayingDeck();
+        if (playingDeck != nullptr) {
+            playingDeck->stop();
+            m_pMpris->notifyPropertyChanged(playerInterfaceName, "Metadata", QVariantMap());
+            m_pausedDeck = playingDeck->group;
+        }
+    } else {
+        for (DeckAttributes* attribute : qAsConst(m_deckAttributes)) {
+            attribute->stop();
+        }
+    }
+}
+
+void MprisPlayer::playPause() {
+    if (autoDjIdle()) {
+        DeckAttributes* playingDeck = findPlayingDeck();
+        if (playingDeck != nullptr) {
+            playingDeck->stop();
+            m_pMpris->notifyPropertyChanged(playerInterfaceName, "Metadata", QVariantMap());
+            m_pausedDeck = playingDeck->group;
+        } else {
+            ControlProxy playing(ConfigKey(m_pausedDeck, "play"));
+            BaseTrackPlayer* player = m_pPlayerManager->getPlayer(m_pausedDeck);
+            DEBUG_ASSERT(player);
+            TrackPointer pTrack = player->getLoadedTrack();
+            playing.set(true);
+        }
+    }
+}
+
+void MprisPlayer::play() {
+    if (!m_pCPAutoDjEnabled->toBool()) {
+        m_pCPAutoDjEnabled->set(true);
+        return;
+    }
+    DeckAttributes* playingDeck = findPlayingDeck();
+    if (playingDeck == nullptr) {
+        ControlProxy playing(ConfigKey(m_pausedDeck, "play"));
+        BaseTrackPlayer* player = m_pPlayerManager->getPlayer(m_pausedDeck);
+        DEBUG_ASSERT(player);
```

Comment:
I had one crash here that I could not reproduce later:

```
3:41:31.777 [Main] debug TrackDAO::updateTrack  TrackDAO: Updating track in database 7726 QFileInfo(/home/poelzi/Music/psyblast/[Omveda Records]/Omveda Records - XIII Monkeys - Psychedelic Renaissance (FREE)/XIII MONKEYS - XIII Monkeys - Psychedelic Renaissance (FREE) - 01 Alien Agenda.flac)
3:41:31.777 [Main] debug mixxx::Logger::debug  SqlTransaction - Started new SQL database transaction on "MIXXX-1"
3:41:32.131 [Main] debug mixxx::Logger::debug  SqlTransaction - Committed SQL database transaction on "MIXXX-1"
3:41:32.131 [Main] debug BaseTrackCache::updateIndexWithQuery  BaseTrackCache(0xb16c2e0) updateIndexWithQuery took 0 ms
3:41:32.131 [Main] debug mixxx::Logger::debug  GlobalTrackCache - Deleting Track(0x7f7ba83fb480)
3:41:32.134 [Main] debug AutoDJProcessor::calculateTransition  player "[Channel2]" calculateTransition()
3:41:32.134 [Controller] debug unknown  CDBG XONE:K2 MIDI 1: outgoing: status 0x9E (ch 15, opcode 0x9), ctrl 0x2D, val 0x7F
3:41:32.134 [Controller] debug unknown  CDBG XONE:K2 MIDI 1: outgoing: status 0x9D (ch 14, opcode 0x9), ctrl 0x2C, val 0x7F
3:41:32.134 [Controller] debug unknown  CDBG XONE:K2 MIDI 1: outgoing: status 0x9C (ch 13, opcode 0x9), ctrl 0x2C, val 0x7F
3:41:32.134 [Controller] debug unknown  CDBG XONE:K2 MIDI 1: outgoing: status 0x9B (ch 12, opcode 0x9), ctrl 0x2E, val 0x7F
3:41:32.134 [Controller] debug unknown  CDBG XONE:K2 MIDI 1: outgoing: status 0x9A (ch 11, opcode 0x9), ctrl 0x2D, val 0x7F
3:41:32.134 [Controller] debug unknown  CDBG XONE:K2 MIDI 1: outgoing: status 0x99 (ch 10, opcode 0x9), ctrl 0x2C, val 0x7F
3:41:32.141 [Main] debug mixxx::Logger::debug  AnalyzerThread - Enqueueing next track 7728
3:41:32.141 [AnalyzerThread 0 #1] debug mixxx::Logger::debug  AnalyzerThread - Dequeued next track 7728
3:41:32.141 [AnalyzerThread 0 #1] debug mixxx::Logger::debug  AnalyzerThread - Analyzing QFileInfo(/home/poelzi/Music/psyblast/[Omveda Records]/Omveda Records - XIII Monkeys - Psychedelic Renaissance (FREE)/XIII MONKEYS - XIII Monkeys - Psychedelic Renaissance (FREE) - 02 Psychedelic Renaissance.flac)
3:41:32.142 [AnalyzerThread 0 #1] debug mixxx::Logger::debug  SoundSourceProxy - SoundSourceProvider "Xiph.org libFLAC" created a SoundSource for file "file:///home/poelzi/Music/psyblast/[Omveda Records]/Omveda Records - XIII Monkeys - Psychedelic Renaissance (FREE)/XIII MONKEYS - XIII Monkeys - Psychedelic Renaissance (FREE) - 02 Psychedelic Renaissance.flac" of type "flac"
3:41:32.142 [AnalyzerThread 0 #1] debug AnalysisDao::loadAnalysesFromQuery  AnalysisDAO fetched 0 analyses, 0 bytes for track 7728 in 0 ms
3:41:32.143 [AnalyzerThread 0 #1] debug AnalyzerEbur128::initialize  Skipping AnalyzerEbur128
3:41:32.143 [AnalyzerThread 0 #1] debug AnalyzerBeats::initialize  AnalyzerBeats preference settings: 
Plugin: "qm-tempotracker:0" 
Min/Max BPM: 70 250 
Fixed tempo assumption: true 
Offset correction: true 
Re-analyze when settings change: false 
Fast analysis: false
3:41:32.143 [AnalyzerThread 0 #1] debug AnalyzerKey::initialize  AnalyzerKey preference settings: 
Plugin: "qm-keydetector:2" 
Re-analyze when settings change: false 
Fast analysis: false
3:41:32.143 [AnalyzerThread 0 #1] debug AnalyzerKey::shouldAnalyze  Track has previous key detection result that is not up to date with latest settings but user preferences indicate we should not re-analyze it.
3:41:36.193 [AnalyzerThread 0 #1] debug Waveform::toByteArray  Writing waveform from byte array: dataSize 371166 allSignalSize 371166 visualSampleRate 441 audioVisualRatio 100
3:41:36.729 [AnalyzerThread 0 #1] debug AnalysisDao::saveAnalysis  AnalysisDAO saved analysis 7007 "3396322 (1601831 compressed)" bytes for track 7728 in 523 ms
3:41:36.729 [AnalyzerThread 0 #1] debug AnalysisDao::saveTrackAnalyses  Saved waveform analysis for trackId 7728 analysisId 7007
3:41:36.729 [AnalyzerThread 0 #1] debug Waveform::toByteArray  Writing waveform from byte array: dataSize 3840 allSignalSize 3840 visualSampleRate 4.5625 audioVisualRatio 9665.76
3:41:36.741 [AnalyzerThread 0 #1] debug AnalysisDao::saveAnalysis  AnalysisDAO saved analysis 7008 "34941 (15475 compressed)" bytes for track 7728 in 11 ms
3:41:36.741 [AnalyzerThread 0 #1] debug AnalysisDao::saveTrackAnalyses  Saved waveform summary analysis for trackId 7728 analysisId 7008
3:41:36.741 [AnalyzerThread 0 #1] debug mixxx::Logger::debug  AnalyzerWaveform - Waveform generation for track 7728 done 4 s
3:41:37.269 [Main] critical  DEBUG ASSERT: "key.isValid() || flags.testFlag(ControlFlag::AllowInvalidKey)" in function ControlProxy::ControlProxy(const ConfigKey &, QObject *, ControlFlags) at /home/poelzi/Projects/mixxx-test/src/control/controlproxy.cpp:19
3:41:37.270 [Main] critical  DEBUG ASSERT: "player" in function void MprisPlayer::play() at /home/poelzi/Projects/mixxx-test/src/broadcast/mpris/mprisplayer.cpp:203
Segmentation fault (core dumped)
```

Reply:
Do you have systemd? If so you can try using coredumpctl to get the backtrace (e. g. `coredumpctl -1 gdb` to open gdb for the last crash).